python/ray/tests/test_component_failures_3.py

```diff
@@ -62,6 +65,75 @@ def ping(self):
         # node must be reconstructed.
         cluster.remove_node(cluster.list_all_nodes()[-1])
 
+# This test covers actor creation node failure scenario with the assumption
+# that the nodes will only fail after the actors are created or reconstructed.
+# NOTE(zhijunfu): This is because right now we only support actor reconstruction
+# after it has been created, and cannot cover worker failure whiling executing
+# actor creatiion task.
+@pytest.mark.skipif(not RAY_FORCE_DIRECT, reason="only test direct actors")
+@pytest.mark.parametrize(
+    "ray_start_cluster", [{
+        "num_cpus": 4,
+        "num_nodes": 3,
+        "do_init": True,
+        "_internal_config": json.dumps({
+            "initial_reconstruction_timeout_milliseconds": 1000,
+            "num_heartbeats_timeout": 100,
+        })
+    }],
+    indirect=True)
+def test_actor_creation_node_failure_after_actor_created(ray_start_cluster):
+    # TODO(swang): Refactor test_raylet_failed, etc to reuse the below code.
+    cluster = ray_start_cluster
+
+    @ray.remote(max_reconstructions=100)
+    class Child(object):
+        def __init__(self, death_probability):
+            self.death_probability = death_probability
+
+        def get_probability(self):
+            return self.death_probability
+
+        def ping(self):
+            # Exit process with some probability.
+            exit_chance = np.random.rand()
+            if exit_chance < self.death_probability:
+                sys.exit(-1)
+
+    num_children = 25
+    # Children actors will die about half the time.
+    death_probability = 0.5
+
+    children = [Child.remote(death_probability) for _ in range(num_children)]
+    while len(cluster.list_all_nodes()) > 1:
+        # Submit some tasks on the actors. About half of the actors will
+        # fail.
+        children_out = [child.ping.remote() for child in children]
+        # Wait a while for all the tasks to complete, so we can know
+        # which actors have died.
+        ready, _ = ray.wait(
+            children_out, num_returns=len(children_out), timeout=5 * 60.0)
+        assert len(ready) == len(children_out)
+
+        # Replace any actors that died.
+        for i, out in enumerate(children_out):
+            try:
+                ray.get(out)
+            except ray.exceptions.RayActorError:
+                children[i] = Child.remote(death_probability)
+
+
+        children_out = [child.get_probability.remote() for child in children]
+        # Wait for new created actors to finish creation before removing a node.
+        # This is needed because right now we don't support reconstructing actors
+        # that died in the process of being created.
+        ready, _ = ray.wait(
+            children_out, num_returns=len(children_out), timeout=5 * 60.0)
+        assert len(ready) == len(children_out)
```

Comment:
what about just adding the following to the original test case? So we don't need to copy-paste a new one.
```
if RAY_FORCE_DIRECT:
   children_out = [child.get_probability.remote() for child in children]
    # ...
```

Reply:
good suggestion. revised.